js/scorm/Connection.js

```diff
@@ -0,0 +1,78 @@
+import Adapt from 'core/js/adapt';
+
+export default class Connection {
+
+  constructor({
+    _isEnabled = false,
+    _silentRetryLimit = 0,
+    _silentRetryDelay = 2000
+  } = {}, ScormWrapper) {
+    this._isEnabled = _isEnabled;
+    this._isInProgress = false;
+    this._isSilentDisconnection = false;
+    this._isDisconnected = false;
+    this._silentRetryLimit = _silentRetryLimit;
+    this._silentRetryDelay = _silentRetryDelay;
+    this._silentRetryTimeout = null;
+    this._silentRetryCount = 0;
+    this._scorm = ScormWrapper;
+  }
+
+  async test() {
+    if (!this._isEnabled || this._isInProgress) return;
+    this._isInProgress = true;
+
+    fetch(`connection.json?nocache=${Date.now()}`)
+    .then(response => {
+      (response?.ok) ? this.onConnectionSuccess() : this.onConnectionError();
+    })
+    .catch(error => {
+      this.onConnectionError();
+    });
```

Comment:
```suggestion
    try {
      const response = await fetch(`connection.json?nocache=${Date.now()}`);
      if (response?.ok) return this.onConnectionSuccess();
    } catch (err) {}
    this.onConnectionError();
```

Reply:
I meant to say: Don't use promises with callbacks when await / async + try/catch is available. Promises and their apis (then/catch) are extremely nesty and incrediblty hard to follow, you'll see eleanor has commented below that it should be further indented with old promise syntax style.

What's the issue with eslint and local errors?

Reply:
await / async is the way of turning promise-based javascript back into that linear, traditional, one line at a time flow, from before promises, instead of handling the flow in callbacks. Much of the way expressions and statements, assignments, errors etc revert to normal, with a few weird exceptions.
```js
[].filter(function predicate (value, index) {

});
```
Filter is a higher-order function, which expects a reference to a non-async predicate function as its first argument. Such that it won't wait if that first argument function references an async function.

```js
[].filter(async function predicate (value, index) {
  let rtnValue = false;
  // do something async
  return rtnValue;
});
```
Filter is expecting the predicate function to return a truthy or a falsey, instead here the predicate function is returning a promise object, as do all async functions. await is the keyword to wait for and extract the value or extract and throw the error from that promise object. Hence why try-catch works with await.
The above code would have to be written using an iteration style that's await/async compatible, like`for`.
```js
const arr = [];
const filteredArr = [];
for (const value of arr) {
  let rtnValue = false;
  // do something async
  if (rtnValue) filteredArr.push(value);
}
```